src/app/app.component.ts

```diff
@@ -307,4 +310,19 @@ export class AppComponent implements OnInit {
   public doHomeAccessCodeError(error: string) {
     this.router.navigateByUrl(`/app/home?acError=${error}`);
   }
+
+  public accessCodeSelectedForDeletion(a: string[]) {
+    this.selectedAccesscodesForDeletion.forEach((element) => {
```

Comment:
Do these lines have an effect?

Is it possible to use only
`this.selectedAccesscodesForDeletion = a;` ?

Reply:
I thought this lines have an effect, because my logic was that adding each element to the a string array, but the way you proposed had the same result. Maybe that had to do with clarity functionality that I wasn't aware of. 
Anyway, I refactored as you proposed.